src/types.ts

```diff
@@ -52,7 +52,9 @@ export type NuxtI18nOptions<Context = unknown> = {
   skipSettingLocaleOnNavigate?: boolean
   // sortRoutes?: boolean
   strategy?: Strategies
-  vueI18n?: I18nOptions | string
+  vueI18n?: {
+    configFile?: string
+  }
```

Comment:
I hope the i18n options keep `I18nOptions | string` because it would break compatibility with v7.

Reply:
It would need to be `string` for this change. Sorry if I'm misunderstanding but why is compatibility with v7 important (since this is another major version)?

Reply:
Indeed, v8 is a major version, so breaking changes are inevitable.
Upgrading to a major version for a large application that has already been built is a huge effort.

Some frameworks (e.g. larave, rails, and react) have been released so that major versions do not cause too much work for the user's application. They are releasing features supported in previous versions in working order, just as Vue3 supports the Options API.

I have once again thought about how much breaking change would detract from the migration experience for the user.

In this PR, it is a change in the configuration interface, so the affected code point is local.
Changing the vueI18n option would not be much of a problem.
Of course, we need to write a migration guide in the documentation.

Reply:
Absolutely - it would be great to have as less breaking changes as possible.

Similar to #1919 also moving some fragile configuration options to other solutions that Nuxt provides for stability, I'm aiming to do the same here. This may have been a small weak-point in v7, so having this solved in v8 would be great. The migration would indeed be documented.